tests/src/FunctionalJavascript/CoursesTest.php

```diff
@@ -46,6 +49,84 @@ public function testCoursesPage() {
+    $anuApp = $assert->waitForElementVisible('css', '#anu-application');
+    $categoryTitles = $anuApp->findAll('css', 'h2');
+    $this->assertNotEmpty($categoryTitles);
+    $this->assertEqual($categoryTitles[1]->getText(), 'Developer guides [DEMO]');
+
+    $courseTitles = $anuApp->findAll('css', '#anu-application h3');
+    $this->assertNotEmpty($courseTitles);
+    $this->assertEqual($courseTitles[1]->getText(), 'Modules, lessons and sections [DEMO]');
+    $courseTitles[1]->click();
+
+    // After clicking on the title it should be inside the course.
+    $courseTitle = $assert->waitForElementVisible('css', '#anu-application h1');
+    $this->assertNotEmpty($courseTitle);
+    $this->assertEqual($courseTitle->getText(), 'Modules, lessons and sections [DEMO]');
+  }
+
+  /**
+   * Test lesson content.
+   */
+  public function testLessons() {
+    $assert = $this->assertSession();
+    $this->drupalGet('lesson/headings');
+
+    // Inside the course. Check the sidebar active element.
+    $this->assertJsCondition('document.querySelector("#anu-application div[data-anu-lms-navigation-item-status=active]").textContent === "Headings"');
+
+    $headingThree = $assert->waitForElementVisible('css', '#anu-application h3');
+    $this->assertNotEmpty($headingThree);
+    $this->assertEqual($headingThree->getText(), 'Lesson heading - h3');
+
+    $next = $assert->waitForElementVisible('css', '[aria-label=Next]');
+    $next->click();
+    $lessonTitle = $assert->waitForElementVisible('css', '#anu-application h4');
+    $this->assertNotEmpty($lessonTitle);
+    $this->assertEqual($lessonTitle->getText(), 'Text');
+
+    $linkInBody = $assert->waitForElementVisible('css', 'a[href="https://github.com/systemseed/anu_lms"]');
+    $this->assertNotEmpty($linkInBody);
+    $this->assertEqual($linkInBody->getText(), 'Link to GitHub repository.');
+
+    $assert->waitForElementVisible('css', '[aria-label=Next]')->click();
+
+    $assert->waitForElementVisible('css', '#anu-application ul li');
+    // Check third item.
+    $this->assertJsCondition('document.querySelector("#anu-application ul li:nth-child(3) div[data-anu-lms-list-item-text]").textContent === "Vanilla"');
+
+    $assert->waitForElementVisible('css', '[aria-label=Next]')->click();
+
+    $image = $assert->waitForElementVisible('css', '#anu-application img');
+    $this->assertNotEmpty($image);
+    $this->assertEqual($image->getAttribute('alt'), 'Image with caption');
+
+    $assert->waitForElementVisible('css', '[aria-label=Next]')->click();
+
+    // Highlights.
+    $highlightHeading = $assert->waitForElementVisible('css', '#anu-application p[data-anu-lms-highlight-heading]');
+    $this->assertNotEmpty($highlightHeading);
+    $this->assertEqual($highlightHeading->getText(), 'Highlight (full width)');
+
+    $assert->waitForElementVisible('css', '[aria-label=Next]')->click();
+    // Dividers.
+    $assert->waitForElementVisible('css', '[aria-label=Next]')->click();
```

Comment:
I assume we will add more checks inside those lessons later

Reply:
I think we are good for the scope of this testing.